test/e2e/run-api-specs-multichain.ts

```diff
@@ -0,0 +1,375 @@
+  ACCOUNT_1,
+} from './helpers';
+import { MultichainAuthorizationConfirmation } from './api-specs/MultichainAuthorizationConfirmation';
+import transformOpenRPCDocument from './api-specs/transform';
+import { MultichainAuthorizationConfirmationErrors } from './api-specs/MultichainAuthorizationConfirmationErrors';
+import { ConfirmationsRejectRule } from './api-specs/ConfirmationRejectionRule';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+const mockServer = require('@open-rpc/mock-server/build/index').default;
+
+async function main() {
+  let testCoverageResults: Call[] = [];
+  const port = 8545;
+  const chainId = 1337;
+
+  const doc = await parseOpenRPCDocument(
+    MultiChainOpenRPCDocument as OpenrpcDocument,
+  );
+  const providerAuthorize = doc.methods.find(
+    (m) => (m as MethodObject).name === 'wallet_createSession',
+  );
+
+  const walletRpcMethods: string[] = [
+    'wallet_registerOnboarding',
+    'wallet_scanQRCode',
+  ];
+  const walletEip155Methods = ['wallet_addEthereumChain'];
+
+  const ignoreMethods = [
+    'wallet_switchEthereumChain',
+    'wallet_getPermissions',
+    'wallet_requestPermissions',
+    'wallet_revokePermissions',
+    'eth_requestAccounts',
+    'eth_accounts',
+    'eth_coinbase',
+    'net_version',
+  ];
+
+  const [transformedDoc, filteredMethods, methodsWithConfirmations] =
+    transformOpenRPCDocument(
+      MetaMaskOpenRPCDocument as OpenrpcDocument,
+      chainId,
+      ACCOUNT_1,
+    );
+  const ethereumMethods = transformedDoc.methods
+    .map((m) => (m as MethodObject).name)
+    .filter((m) => {
+      const match =
+        walletRpcMethods.includes(m) ||
+        walletEip155Methods.includes(m) ||
+        ignoreMethods.includes(m);
+      return !match;
+    });
+  const confirmationMethods = methodsWithConfirmations.filter(
+    (m) => !ignoreMethods.includes(m),
+  );
+  const scopeMap: Record<InternalScopeString, string[]> = {
+    [`eip155:${chainId}`]: ethereumMethods,
+    'wallet:eip155': walletEip155Methods,
+    wallet: walletRpcMethods,
+  };
+
+  const reverseScopeMap = Object.entries(scopeMap).reduce(
+    (acc, [scope, methods]: [string, string[]]) => {
+      methods.forEach((method) => {
+        acc[method] = scope;
+      });
+      return acc;
+    },
+    {} as { [method: string]: string },
+  );
+
+  // Multichain API excluding `wallet_invokeMethod`
+  await withFixtures(
+    {
+      dapp: true,
+      fixtures: new FixtureBuilder().build(),
+      disableGanache: true,
+      title: 'api-specs-multichain coverage',
+    },
+    async ({
+      driver,
+      extensionId,
+    }: {
+      driver: Driver;
+      extensionId: string;
+    }) => {
+      await unlockWallet(driver);
+
+      // Navigate to extension home screen
+      await driver.navigate(PAGES.HOME);
+
+      // Open Dapp
+      await openDapp(driver, undefined, DAPP_URL);
+
+      // fix the example for wallet_createSession
```

Comment:
Is this something we can do now?

Reply:
yeah probably could at this point. We were probably waiting for the API to steady itself, but seems pretty locked in now.

Reply:
here's a PR to make this update on the API-Specs: https://github.com/MetaMask/api-specs/pull/279

Reply:
Resolved here https://github.com/MetaMask/metamask-extension/pull/27782/commits/1e20517c0c66f381f519aa1b3c10b826cd017c9b